components/camel-http/src/main/java/org/apache/camel/component/http/HttpMethods.java

```diff
@@ -41,38 +41,32 @@
     OPTIONS(HttpOptions.class),
     TRACE(HttpTrace.class);
 
-    final Class<? extends HttpRequestBase> clazz;
     final boolean entity;
 
     HttpMethods(Class<? extends HttpRequestBase> clazz) {
-        this.clazz = clazz;
         entity = HttpEntityEnclosingRequestBase.class.isAssignableFrom(clazz);
     }
 
     public HttpRequestBase createMethod(final String url) {
-        try {
-            switch (this) {
-            case GET:
-                return new HttpGet(url);
-            case PATCH:
-                return new HttpPatch(url);
-            case POST:
-                return new HttpPost(url);
-            case PUT:
-                return new HttpPut(url);
-            case DELETE:
-                return new HttpDelete(url);
-            case HEAD:
-                return new HttpHead(url);
-            case OPTIONS:
-                return new HttpOptions(url);
-            case TRACE:
-                return new HttpTrace(url);
-            default:
-                throw new RuntimeException("no such method " + this);
-            }
-        } catch (IllegalArgumentException e) {
-            throw new RuntimeException(e);
+        switch (this) {
+        case GET:
+            return new HttpGet(url);
+        case PATCH:
+            return new HttpPatch(url);
+        case POST:
+            return new HttpPost(url);
+        case PUT:
+            return new HttpPut(url);
+        case DELETE:
+            return new HttpDelete(url);
+        case HEAD:
+            return new HttpHead(url);
+        case OPTIONS:
+            return new HttpOptions(url);
+        case TRACE:
+            return new HttpTrace(url);
+        default:
+            throw new RuntimeException("no such method " + this);
```

Comment:
Perhaps?
```suggestion
            throw new IllegalArgumentException("Unupported HTTP verb: " + this);
```

Reply:
Yes, that was another part of https://github.com/apache/camel/pull/3469/commits/ceffe629e802090e138bdc85f719523d8d4961de I considered changing.

But the exception will only be thrown if another constant is added to the `HttpMethods` enum and no matching case is added to this switch (or am I missing something?). I was not sure which exception class to use. My first thought was `llegalArgumentException` as well, but as the switch is on `this` and not on an argument of the `createMethod` method so using an `llegalArgumentException` seemed misleading. An exception signalling that a bug (missing implementation) was encountered seems more appropriate (like `GroovyBugError` used by Groovy).

Reply:
Yeah, I'm nitpicking, there's little chance for that exception to even occur. IllegalArgumentException or IllegalStateException are my go-to exceptions in lieu of RuntimeException. Not that important, only of you think it merits the change...

Reply:
Replacing the swith with inheritance (see: https://github.com/apache/camel/pull/3475) would allow us to get rid of the exception.